abci/example/kvstore/kvstore.go

```diff
@@ -28,15 +28,19 @@ type State struct {
 }
 
 func loadState(db dbm.DB) State {
-	stateBytes := db.Get(stateKey)
 	var state State
-	if len(stateBytes) != 0 {
-		err := json.Unmarshal(stateBytes, &state)
-		if err != nil {
-			panic(err)
-		}
-	}
 	state.db = db
+	stateBytes, err := db.Get(stateKey)
+	if err != nil {
+		panic(err)
+	}
+	if stateBytes == nil {
```

Comment:
```suggestion
	if len(stateBytes) != 0 {
```

!= nil and len(...) == 0 are not the same things